spec/integration/runner/variant-reassignment-example-2.spec.js

```diff
@@ -0,0 +1,92 @@
+ * |                           | product-type: "pt1"                         |                    | product-type: "pt1"                                           |
+ * |                           | variants: v3, v4                            |                    | variants: v4                                                  |
+ * +---------------------------+---------------------------------------------+                    +---------------------------------------------------------------+
+ * |                           |                                             |                    | Product:                                                      |
+ * |                           |                                             |                    | id: "3"                                                       |
+ * |                           |                                             |                    | slug: { en: "product-1", de: "produkte-1" }                   |
+ * |                           |                                             |                    | product-type: "pt2"                                           |
+ * |                           |                                             |                    | variants: v1, v3                                              |
+ * +---------------------------+---------------------------------------------+--------------------+---------------------------------------------------------------+
+ */
+/* eslint-enable max-len */
+describe('Variant reassignment', () => {
+  const logger = utils.createLogger(__filename)
+  let ctpClient
+  let product1
+  let product2
+
+  before(async () => {
+    ctpClient = await utils.createClient()
+    const productType = await utils.ensureProductType(ctpClient)
+    const productDraft1 = utils.generateProduct(['1', '2'], productType.id)
+    product1 = await utils.ensureResource(ctpClient.products, productDraft1)
+    const productDraft2 = utils.generateProduct(['3', '4'], productType.id)
+    product2 = await utils.ensureResource(ctpClient.products, productDraft2)
+  })
+
+  after(() =>
+    utils.deleteResourcesAll(ctpClient, logger)
+  )
+
+  it('moving variant v3 + removing variant v2 + changing productType', async () => {
+    productDraftProductType.name = 'product-draft-product-type'
+    const productType = await utils.ensureResource(ctpClient.productTypes, productDraftProductType)
+
+    const reassignment = new VariantReassignment([], logger, {})
+    await reassignment.execute([{
+      productType: {
+        id: productType.id
+      },
+      key: 'sample-product1',
+      name: {
+        en: 'Sample product1'
+      },
+      slug: {
+        en: 'sample-product1'
+      },
+      masterVariant: {
+        sku: '1',
+        prices: []
+      },
+      variants: [
+        {
+          sku: '3',
+          prices: []
+        }
+      ]
+    }], [product1, product2])
```

Comment:
I think this should be the products that are passed to product-import from the user's apps.

Reply:
Because when we create products here in tests we return a result of `products.create()` which is different from products fetched using `products.fetch()` and returned in product-import

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/8/commits/92a56749a258b5e4f850673e484fd52686d134b2